addons/event/tests/test_event_internals.py

```diff
@@ -714,6 +714,37 @@ def test_event_seats(self):
 @tagged('event_registration')
 class TestEventRegistrationData(TestEventInternalsCommon):
 
+    @users('user_eventmanager')
+    def test_registration_date_closed(self):
+        """Test changes in date_closed field when state is changed."""
+        with self.mock_datetime_and_now('2025-05-03 17:00:00'):
+            event = self.env['event.event'].create({
+                'name': 'Test Event',
+                'date_begin': FieldsDatetime.to_string(datetime.now()),
+                'date_end': FieldsDatetime.to_string(datetime.now() + timedelta(days=2)),
+            })
+            attendee = self.env['event.registration'].create({
+                'name': 'Test Registration',
+                'event_id': event.id,
+                'state': 'done',
+            })
+            self.assertEqual(attendee.date_closed, datetime.now())
+
+            attendee.action_set_done()
+            message = '<p>Attended on 03/05/2025</p>'
+            self.assertTrue(message in attendee.message_ids.mapped('body'),
+                'Expected a "Attended on 03/05/2025" message in the chatter.')
+
+            attendee.action_set_done()
+            self.assertEqual(attendee.message_ids.mapped('body').count(message), 2,
+                'Logged message when marked as attended again.')
+
+        with self.mock_datetime_and_now('2025-05-04 17:00:00'):
+            attendee.action_set_done()
+            new_message = '<p>Attended on 04/05/2025</p>'
+            self.assertTrue(new_message in attendee.message_ids.mapped('body'),
+                'Expected a "Attended on 04/05/2025" message in the chatter.')
```

Comment:
Sorry I mixed up the changes (thought message could change for some reason, which makes no sense). 
No need to check this twice actually I don't think we gain anything ^^'
Feel free to tell me if you think my comments don't make sense :p